nsv13/code/modules/overmap/knpc.dm

```diff
@@ -376,7 +406,13 @@ This is to account for sec Ju-Jitsuing boarding commandos.
 	var/mob/living/carbon/human/H = HA.parent
 	var/obj/item/storage/S = H.back
 	var/obj/item/gun/target_item = null
-	//Okay first off, is the gun already on our person?
+	//We must have lost our gun somehow, get it from the floor if we simply dropped it.
+	if(istype(H.loc, /turf))
+		var/turf/T = H.loc
+		target_item = locate(/obj/item/gun) in T.contents
+		if(H.put_in_hands(target_item))
+			return TRUE
+	//Otherwise, is there a gun already on our person?
```

Comment:
Potentially loops forever if we just intentionally dropped our gun (e.g. because out of ammo) assuming there is another valid weapon somewhere on the ground nearby.